pandas/io/tests/test_parsers.py

```diff
@@ -2636,7 +2630,145 @@ def test_eof_states(self):
                           StringIO(data), escapechar='\\')
 
 
-class TestPythonParser(ParserTests, tm.TestCase):
+class CompressionTests(object):
+    def test_zip(self):
```

Comment:
need tests that exercises ``infer`` (the default). you can do this in each individual test for that zip where you name a file with the appropriate extension and don't pass the compression option.

Its possible tests like these are elsewhere, not sure.